### PR TITLE
admin/status: test GPG signature check

### PR DESCRIPTION
**Pull request by r4f4**
This commit complements PR #1787 

Signed-off-by: Rafael Fonseca <r4f4rfs@gmail.com>

### CODE REVIEW

File: Makefile-tests.am [hunk 1]
```diff
@@ -109,6 +109,7 @@ _installed_or_uninstalled_test_scripts = \
 	tests/test-admin-pull-deploy-split.sh \
 	tests/test-admin-locking.sh \
 	tests/test-admin-deploy-clean.sh \
+    tests/test-admin-gpg.sh \
```
Comment (rfairley) [inline Makefile-tests.am]: Minor: looks like a tab is needed here to indent this line

**Comment by rfairley** [comment]
@rh-atomic-bot retry

**Comment by rfairley** [comment]
bot, retest this please

File: tests/test-admin-gpg.sh [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+# Exports OSTREE_SYSROOT so --sysroot not needed.
+setup_os_repository_signed "archive" "syslinux"
+
+echo "1..2"
+
+${CMD_PREFIX} ostree --repo=sysroot/ostree/repo remote add testos $(cat httpd-address)/ostree/testos-repo
+${CMD_PREFIX} ostree --repo=sysroot/ostree/repo pull-local --gpg-verify=true --remote=testos testos-repo testos/buildmaster/x86_64-runtime
+# This initial deployment gets kicked off with some kernel arguments
+${CMD_PREFIX} ostree admin deploy --karg=root=LABEL=MOO --karg=quiet --os=testos testos:testos/buildmaster/x86_64-runtime
```
Comment (cgwalters) [inline tests/test-admin-gpg.sh]: To extend this test in the future we should add an upgrade and then test the `ostree admin upgrade` path so we're exercising the GPG pull via that code too (which is mostly the same thing in practice).

File: tests/test-admin-gpg.sh [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+	      ostree_repo_init testos-repo --mode=${mode}
+    else
+	      ostree_repo_init testos-repo
+    fi
+
+    cd ${test_tmpdir}
+    mkdir osdata
+    cd osdata
+    kver=3.6.0
+    mkdir -p usr/bin ${bootdir} usr/lib/modules/${kver} usr/share usr/etc
```
Comment (cgwalters) [inline tests/test-admin-gpg.sh]: Future enhancement: Let's avoid duplicating the code for setup (maybe make gpg signatures a flag)?

**Comment by cgwalters** [comment]
@rh-atomic-bot r+ 54ae4d1

File: tests/test-admin-gpg.sh (hunk 3 above)
Comment (rfairley) [reply tests/test-admin-gpg.sh]: :+1: Agree - my thought was for this test it'd be okay as `setup_os_repository_signed` is local to this file, but to enhance in future, the one in `libtest.sh` could be updated with a flag.

**Comment by dustymabe** [comment]
thanks @r4f4 !

**Pull request closed without merging**